lib/utils/addEntries.js

```diff
@@ -120,16 +120,37 @@ function addEntries(compiler, options, server) {
   // eslint-disable-next-line no-shadow
   compilers.forEach((compiler) => {
     const config = compiler.options;
+    const target = config.target;
     /** @type {Boolean} */
-    const webTarget = [
-      'web',
-      'webworker',
-      'electron-renderer',
-      'node-webkit',
-      // eslint-disable-next-line no-undefined
-      undefined,
-      null,
-    ].includes(config.target);
+    let webTarget = false;
+    try {
+      // only available in webpack@5
+      const {
+        getTargetsProperties,
+        getTargetProperties,
+        // @ts-ignore
+        // eslint-disable-next-line import/no-unresolved
+      } = require('webpack/lib/config/target');
+      if (target) {
+        const targetProperties =
+          typeof target === 'string'
+            ? getTargetProperties(target, config.context)
+            : getTargetsProperties(target, config.context);
+        // optimistic
+        webTarget = targetProperties.web || targetProperties.web === null;
```

Comment:
Should this be optimistic, i.e., include entries if targets are mixed, e.g., browser and Node?

Reply:
What about move this logic in plugin too, ideally we should rely on https://github.com/webpack/webpack/blob/master/lib/config/target.js#L153, so it allows to webpack adding new targets without updating our code?

Reply:
This uses the same logic with webpack@5 (`getTargetProperties` and `getTargetsProperties`).

Reply:
Will be great to avoid this, because it is internal and logic can be changed in any time (we try to avoid it, but sometime it happens)

Reply:
Anyway why do not put this logic in the plugin? 

In future webpack-dev-server should be refactored as plugin, so we still have to do it in the future

Reply:
@evilebottnawi By plugin, do you mean by putting this in `DevServerEntryPlugin` and plugin doing nothing if web targets are detected?

Reply:
I mean:
- let's rename `DevServerEntryPlugin` to `DevServerPlugin`
- put logic for `hot`, `target`, `entries` (I think most of lines in this file) in this plugin, it is more flexibility